packages/react-shopify-app-bridge/src/Provider.tsx

```diff
@@ -132,12 +121,9 @@ const InnerGadgetProvider = memo(
         gadgetAppUrl,
       });
 
-      if (isEmbedded && appBridge) {
-        Redirect.create(appBridge).dispatch(Redirect.Action.REMOTE, redirectURLWithOAuthParams);
-      } else {
-        window.location.assign(redirectURLWithOAuthParams);
-      }
-    }, [appBridge, gadgetAppUrl, isEmbedded, isRootFrameRequest, originalQueryParams, runningShopifyAuth]);
+      open(redirectURLWithOAuthParams, "_top");
```

Comment:
Changed this to ensure it's a top level redirect when embedded and it's an embedded app type